python/audit/utils.py

```diff
@@ -86,6 +91,70 @@ def _get_deleted_obj(view: View, request: Request, kwargs) -> Model | None:
     return deleted_obj
 
 
+def _get_object_changes(prev_data: dict, current_obj: Model) -> dict:
+    serializer_class = None
+    if isinstance(current_obj, Group):
+        serializer_class = GroupAuditSerializer
+    elif isinstance(current_obj, Role):
+        serializer_class = RoleAuditSerializer
+    elif isinstance(current_obj, User):
+        serializer_class = UserAuditSerializer
+    elif isinstance(current_obj, Policy):
+        serializer_class = PolicyAuditSerializer
+
+    if not serializer_class:
+        return {}
+
+    current_data = serializer_class(current_obj).data
+    current_fields = {k: v for k, v in current_data.items() if prev_data[k] != v}
+    if not current_fields:
+        return current_fields
+
+    object_changes = {
+        "current": current_fields,
+        "previous": {k: v for k, v in prev_data.items() if k in current_fields},
+    }
+    if object_changes["current"].get("password"):
+        object_changes["current"]["password"] = "******"
+
+    if object_changes["previous"].get("password"):
+        object_changes["previous"]["password"] = "******"
+
+    return object_changes
+
+
+def _get_obj_changes_data(view: ModelViewSet) -> tuple[dict | None, Model | None]:
+    prev_data = None
+    current_obj = None
+    serializer_class = None
+    model = None
+    if (
+        isinstance(view, ModelViewSet)
+        and view.action in {"update", "partial_update"}
+        and view.kwargs.get("pk")
+    ):
+        if view.__class__.__name__ == "GroupViewSet":
+            serializer_class = GroupAuditSerializer
+            model = Group
+        elif view.__class__.__name__ == "RoleViewSet":
+            serializer_class = RoleAuditSerializer
+            model = Role
+        elif view.__class__.__name__ == "UserViewSet":
+            serializer_class = UserAuditSerializer
+            model = User
+        elif view.__class__.__name__ == "PolicyViewSet":
+            serializer_class = PolicyAuditSerializer
+            model = Policy
+
+        if serializer_class:
+            current_obj = model.objects.filter(pk=view.kwargs["pk"]).first()
+            prev_data = serializer_class(model.objects.filter(pk=view.kwargs["pk"]).first()).data
+            if current_obj:
```

Comment:
`current_obj` may be `ModelObject` or `None`. Maybe use `if current_obj is not None:` ?